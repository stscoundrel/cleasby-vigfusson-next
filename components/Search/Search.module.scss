.form {
  display: flex;
}

.input {
  margin-right: 5px;
}