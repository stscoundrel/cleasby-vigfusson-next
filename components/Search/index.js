import { useState } from 'react'
import { useRouter } from 'next/router'

export default function Search() {
  const [search, setSearch] = useState('')
  const router = useRouter()

  const handleSearch = (e) => {
    e.preventDefault()
    router.push(`/search/${search}`)
  }

  return (
   <form onSubmit={(e) => handleSearch(e)}>
      <input type="text" name="search" value={search} onChange={(e) => setSearch(e.target.value)}/>
      <button type="submit">Search</button>
    </form>
  )
}
