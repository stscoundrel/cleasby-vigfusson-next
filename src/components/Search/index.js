import { useState } from 'react'
import { useRouter } from 'next/router'
import styles from './Search.module.scss'

export default function Search() {
  const [search, setSearch] = useState('')
  const router = useRouter()

  const handleSearch = (e) => {
    e.preventDefault()
    router.push(`/search/?query=${search}`)
  }

  return (
   <form className={styles.form} onSubmit={(e) => handleSearch(e)}>
      <input className={styles.input} aria-label="Search" type="search" name="search" value={search} onChange={(e) => setSearch(e.target.value)}/>
      <button className="button button--secondary button--no-margs" name="submit" type="submit">Search</button>
    </form>
  )
}
