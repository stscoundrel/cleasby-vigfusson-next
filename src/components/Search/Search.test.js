import ReactDOM from 'react-dom/client'
import renderer from 'react-test-renderer'
import Search from './index'

const mockHandler = jest.fn()
const mockEvent = jest.fn()
mockEvent.preventDefault = jest.fn()

/**
 * Mock router
 */
jest.mock('next/router', () => ({
  useRouter() {
    return {
      locale: undefined,
      defaultLocale: undefined,
      asPath: '/test',
      push: mockHandler,
    }
  },
}))

describe('Search component', () => {
  test('Does not crash', () => {
    const div = document.createElement('div')
    const root = ReactDOM.createRoot(div)
    root.render(<Search />)
  })

  test('Matches snapshot', () => {
    const tree = renderer.create(<Search />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Triggers router action when submitted', () => {
    const tree = renderer.create(<Search />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Triggers callback after click', async () => {
    const tree = renderer.create(
      <Search />,
    )

    // Submit search form.
    await renderer.act(async () => {
      expect(mockHandler).not.toHaveBeenCalled()

      await tree.root.findByType('form').props.onSubmit(mockEvent)

      // Assert mockrouter received a push.
      expect(mockHandler).toHaveBeenCalled()
      expect(mockHandler).toHaveBeenCalledWith('/search/?query=')
      expect(mockHandler.mock.calls.length).toBe(1);
    })
  })
})
