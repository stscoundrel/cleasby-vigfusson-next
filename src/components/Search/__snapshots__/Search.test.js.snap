// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component Matches snapshot 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <input
    aria-label="Search"
    className="input"
    name="search"
    onChange={[Function]}
    type="search"
    value=""
  />
  <button
    className="button button--secondary button--no-margs"
    name="submit"
    type="submit"
  >
    Search
  </button>
</form>
`;

exports[`Search component Triggers router action when submitted 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <input
    aria-label="Search"
    className="input"
    name="search"
    onChange={[Function]}
    type="search"
    value=""
  />
  <button
    className="button button--secondary button--no-margs"
    name="submit"
    type="submit"
  >
    Search
  </button>
</form>
`;
